packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -145,62 +145,24 @@ const DataGrid = injectIntl((props) => {
     rowHeaderIndex,
   } = props;
 
-  if (pinnedColumns.length === 0) {
-    // eslint-disable-next-line no-console
-    console.warn(ERRORS.PINNED_COLUMNS_UNDEFINED);
-  }
-
-  // Default column size constraints
-  const defaultColumnMinimumWidth = 60;
-  const defaultColumnMaximumWidth = 300;
-
-  const [pinnedColumnOffsets, setPinnedColumnOffsets] = useState([0]);
-
-  // Initialize column width properties
-  const initializeColumn = (column) => ({
-    ...column,
-    width: column.width || defaultColumnWidth,
-    minimumWidth: column.minimumWidth || defaultColumnMinimumWidth,
-    maximumWidth: column.maximumWidth || defaultColumnMaximumWidth,
-  });
-
-  const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(pinnedColumns).concat(overflowColumns);
-  const [dataGridColumns, setDataGridColumns] = useState(displayedColumns.map((column) => initializeColumn(column)));
-
-  // Manage column resize
-  const [tableHeight, setTableHeight] = useState(0);
-  const [activeIndex, setActiveIndex] = useState(null);
-  const activeColumnPageX = useRef(0);
-  const activeColumnWidth = useRef(200);
-  const tableWidth = useRef(0);
-
-  const grid = useRef();
+  const tableRef = useRef();
   const gridContainerRef = useRef();
 
   const handleFocus = useRef(true);
 
-  const [focusedRow, setFocusedRow] = useState(0);
-  const [focusedCol, setFocusedCol] = useState(0);
+  const focusedRow = useRef(0);
+  const focusedCol = useRef(0);
```

Comment:
This should not have been changed to use a ref instead of state variable.  It alters the entire implementation.

Reply:
I know this isn't ideal, but I have not been able to get the focus logic working unless we move the tracking from state to refs. This also gets us closest to the original behavior of the WDG, DG and Flowsheet.

Reply:
I am pretty sure that we have logic that depends on this, especially in the upcoming PRs.  We should be able to make the logic work without making this change.  How did it work before?